File: app/src/main/java/com/example/xc_m6_java/database/UserRepository.java
package com.example.xc_m6_java.database;

import android.app.Application;

import com.example.xc_m6_java.manager.RoomManager;
import com.example.xc_m6_java.model.User;

import java.util.List;

public class UserRepository {

    private UserDao userDao;

    public UserRepository(Application application) {
        RoomManager db = RoomManager.getDatabase(application);
        userDao = db.usersDao();
    }

    public List<User> getUsers() {
        return userDao.getUsers();
    }

    public void saveUser(User user) {
        userDao.saveUser(user);
    }
}

File: app/src/main/java/com/example/xc_m6/database/UserRepository.kt
package com.example.xc_m6.database

import android.app.Application
import com.example.xc_m6.manager.RoomManager
import com.example.xc_m6.model.User

class UserRepository {

    lateinit var userDao: UserDao

    constructor(application: Application) {
        val db = RoomManager.getDatabase(application)
        userDao = db.userDao()
    }

    fun getUsers(): List<User> {
        return userDao.getUsers()
    }

    fun saveUser(user: User) {
        userDao.saveUser(user)
    }
}